runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -314,24 +417,35 @@ private Batcher() {
     @Override
     public boolean commitWorkItem(
         String computation, WorkItemCommitRequest commitRequest, Consumer<CommitStatus> onDone) {
-      if (!canAccept(commitRequest.getSerializedSize() + computation.length())) {
+      if (!canAccept(commitRequest.getSerializedSize() + computation.length())
+          || hasReceivedShutdownSignal()) {
```

Comment:
can we remove the shutdown check here? seems easier to just handle it later.  Otherwise if we return false here, caller thinks batcher is full and  new batchers might be created and repeat.